### PR TITLE
 Only create one __init__.py per project, to prevent race conditions  

### PR DESCRIPTION
**Pull request by JonathanMeans**
I've verified through prints that python_package(), when called from SIX, only gets called twice (coda and pysix). If I'm right about what the problem is, this will fix it. Don't know how to be more sure, short of re-installing several dozen times.

### CODE REVIEW

**Comment by JonathanMeans** [comment]
This is ready to go with `bld.get_tgen_by_name()`

File: build/build.py [hunk 1]
```diff
@@ -499,10 +499,13 @@ def swigModule(self, **modArgs):
             # this turns the folder at the destination path into a package
 
             initTarget = init_tgen_name
-            bld(features = 'python_package',
-                name = initTarget,
-                target='__init__.py',
-                install_path = installPath)
+            try:
```
Comment (asylvest) [inline build/build.py]: OK cool - glad this worked.  This is going to be non-obvious when someone first looks at it (for example, in six months I will totally forget why we did it this way).  Can you add some comments explaining a) what get_tgen_by_name() is doing and why it's in a try/catch and b) what fundamentally we're doing here in terms of trying to get this target created exactly once but then reused across the targets with the same package name?

Comment (JonathanMeans) [inline build/build.py]: I've added the comments.

**Comment by asylvest** [comment]
Looks good - thanks for getting this fixed!

FYI @apakanati

**Pull request merged**